spec/claCheckSpec.js

```diff
@@ -0,0 +1,212 @@
+// limitations under the License.
+
+/**
+ * @fileoverview Spec for CLA Check Github Action.
+ */
+
+const claCheckGithubActionModule =
+ require('../actions/src/cla_check/claCheck.js');
+const core = require('@actions/core');
+const dispatcher = require('../actions/src/dispatcher');
+const github = require('@actions/github');
+const { google } = require('googleapis');
+const { OAuth2Client } = require('google-auth-library');
+const pullRequestPayload = require('../fixtures/pullRequestPayload.json');
+const sheetsToken = JSON.stringify(require('../fixtures/token.json'));
+const sheetsCredentials =
+  JSON.stringify(require('../fixtures/credentials.json'));
+
+
+describe('CLA check github action Module', () => {
+  beforeEach(async () => {
+    process.env.SHEETS_CRED = sheetsCredentials;
+    process.env.SHEETS_TOKEN = sheetsToken;
+    process.env.GITHUB_TOKEN = 'sample-token';
+    github.context.eventName = 'pull_request';
+    github.context.payload = pullRequestPayload.payload;
+
+    octokit = {
+      issues: {
+        createComment: jasmine.createSpy('createComment').and.resolveTo({})
+      },
+    };
+
+    Object.setPrototypeOf(github.GitHub, function () {
+      return octokit;
+    });
+
+    spyOnProperty(github.context, 'repo').and.returnValue({
+      owner: pullRequestPayload.payload.repository.owner.login,
+      repo: pullRequestPayload.payload.repository.name,
+    });
+    spyOn(claCheckGithubActionModule, 'claCheckGithubAction').and.callThrough();
+  });
+
+  describe('called for only PR opened or reopened action', () => {
+    beforeEach(() => {
+      spyOn(google, 'sheets').and.returnValue({
+        spreadsheets: {
+          values: {
+            get: jasmine.createSpy('get').and.callFake(async (obj, cb) => {
+              await cb(null, {
+                data: {
+                  values: [['testuser7777']]
+                },
+              });
+            }),
+          },
+        },
+      });
+    });
+
+    it('should be called for PR opened action', async () => {
+      await dispatcher.dispatch('pull_request', 'opened');
+
+      expect(claCheckGithubActionModule.claCheckGithubAction)
+        .toHaveBeenCalled();
+    });
+
+    it('should be called for PR reopened action', async () => {
+      await dispatcher.dispatch('pull_request', 'reopened');
+
+      expect(claCheckGithubActionModule.claCheckGithubAction)
+        .toHaveBeenCalled();
+    });
+  });
+
+  describe('user has signed cla sheet', () => {
+    beforeEach(function () {
+      spyOn(core, 'info');
+      spyOn(google, 'sheets').and.returnValue({
+        spreadsheets: {
+          values: {
+            get: jasmine.createSpy('get').and.callFake(async (obj, cb) => {
+              await cb(null, {
+                data: {
+                  values: [['testuser7777']]
+                },
+              });
+            }),
+          },
+        },
+      });
+    });
+
```

Comment:
You should also assert that it does not create the cla comment.

Reply:
DONE